### PR TITLE
fix: error accessing price and identifier when build product and cate…

### PR DESCRIPTION
**Pull request by birnayt**
## What's the purpose of this pull request?
<em>Fix the error of trying to access the price and identifier when the API does not return these values.</em>

## How it works? 
<em>Just put a optional chaining in places where the value doesn't always exist </em>

## How to test it?
<em>Try using a marketplace like "homologappoffpremium" without that fix and run the build two or three times.</em> 

erro-build

### CODE REVIEW

**Review by icazevedo — APPROVED** [review]
LGTM. This doesn't fix the root of the problem but while we don't find a more permanent fix to the issue, we shouldn't have problems building the store.

**Review by igorbrasileiro — APPROVED** [review]

**Review by victorhmp — APPROVED** [review]

File: src/components/sections/ProductDetails/ProductDetails.tsx [hunk 1]
```diff
@@ -34,13 +34,13 @@ function ProductDetails({ product: staleProduct }: Props) {
...
+      offers: { offers },
     },
   } = data
 
-  const price = offers.offers[0]?.price
-  const listPrice = offers.offers[0]?.listPrice
-  const seller = offers.offers[0]?.seller
+  const price = offers[0]?.price
+  const listPrice = offers[0]?.listPrice
+  const seller = offers[0]?.seller
```
Comment (igorbrasileiro) [inline src/components/sections/ProductDetails/ProductDetails.tsx:43]: Can you explain the motivation behind those changes? IMO, it seems to be the same, semantically and for readability.

**Review by igorbrasileiro — CHANGES_REQUESTED** [review]

File: src/components/sections/ProductDetails/ProductDetails.tsx (hunk 1 above)
Comment (birnayt) [reply src/components/sections/ProductDetails/ProductDetails.tsx:43]: The motivation was just to improve reading.

File: src/components/product/ProductCard/ProductCard.tsx [hunk 2]
```diff
@@ -41,7 +41,9 @@ function ProductCard({ product, index, className }: Props) {
     return lowestPriceOffer
   }, [spotPrice, offers])
 
-  const { listPrice, seller } = offers[selectedOffer]
+  const selectedOfferPrice = offers[selectedOffer]
+  const listPrice = selectedOfferPrice?.listPrice
+  const seller = selectedOfferPrice?.seller
```
[the remark this answers is not in the document]
Comment (birnayt) [reply src/components/product/ProductCard/ProductCard.tsx:46]: Hmmm thanks, didn't know about this one. I will fix.

**Pull request closed without merging**